app/models/rook.rb

```diff
@@ -2,4 +2,11 @@ class Rook < Piece
   def to_unicode
     black? ? "\u265C" : "\u2656"
   end
+
+  def valid_move?(destination_x, destination_y)
+    distance_x = (destination_x - x_position).abs
+    distance_y = (destination_y - y_position).abs
+
+    distance_x == distance_y || distance_x.zero? || distance_y.zero?
```

Comment:
This should be `distance_x.zero? || distance_y.zero?` because what you currently have is the queen's valid move method
Also you need to add this in factories.rb `factory :rook do 
association :game 
end` this should solve your rspec failing problem

Reply:
The additional distance portion I had must have been part of the queen's diagonal move? I was not aware what that covered. I currently do have that in factories.rb and doesn't appear to be related to the error that we are still getting:
```
An error occurred while loading ./spec/models/bishop_spec.rb.
Failure/Error: game = FactoryBot.create(:game)

ActiveRecord::RecordInvalid:
  Validation failed: Email has already been taken
```
Displays at the end: `0 examples, 0 failures, 1 error occurred outside of examples`